docs/tutorials/getting-started-with-csharp/working-with-linq.md

```diff
@@ -0,0 +1,510 @@
+    foreach (var c in startingDeck)
+        Console.WriteLine(c);
+        
+    Console.WriteLine();
+    var times = 0;
+    var shuffle = startingDeck;
+    do
+    {
+        //shuffle = shuffle.Take(26).LogQuery("Top Half")
+        //    .InterleaveSequenceWith(shuffle.Skip(26).LogQuery("Bottom Half")).LogQuery("Shuffle");
+
+        shuffle = shuffle.Skip(26).LogQuery("Bottom Half")
+            .InterleaveSequenceWith(shuffle.Take(26).LogQuery("Top Half")).LogQuery("Shuffle");
+
+        foreach (var c in shuffle)
+            Console.WriteLine(c);
+        times++;
+        Console.WriteLine(times);
+    } while (!startingDeck.SequenceEquals(shuffle));
+    Console.WriteLine(times);
+}
+```
+
+Notice that you don't log every time you access a query. You log only when you create
+the original query. The program still takes a long time to run, but now you can see why.
+If you run out of patience running the outer shuffle with logging turned on, switch back
+to the inner shuffle. You'll still see the lazy evaluation efects. In one run, it executes
+2592 queries, including all the value and suit generation.
+
+There is an easy way to update this program to avoid all those executions. There are 
+LINQ methods `ToArray()` and `ToList()` that cause the query to run, and store the results
+in an array or a list, respectively. You use these methods to cache the data results of a query
+rather than execute the source query again.  Append the queries that generate the card decks
+with a call to `Toarray()` and run the query again:
+
+```cs
+public static void Main(string[] args)
+{
+var startingDeck = (from s in Suits().LogQuery("Suit Generation")
+                    from v in Ranks().LogQuery("Rank Generation")
+                    select new { Suit = s, Rank = r })
+                    .LogQuery("Starting Deck")
+                    .ToArray();
+    foreach (var c in startingDeck)
+        Console.WriteLine(c);
+        
+    Console.WriteLine();
+    var times = 0;
+    var shuffle = startingDeck;
+    do
+    {
+        shuffle = shuffle.Take(26).LogQuery("Top Half")
+            .InterleaveSequenceWith(shuffle.Skip(26).LogQuery("Bottom Half")).LogQuery("Shuffle").ToArray();
+
+        //shuffle = shuffle.Skip(26).LogQuery("Bottom Half")
+        //    .InterleaveSequenceWith(shuffle.Take(26).LogQuery("Top Half")).LogQuery("Shuffle");
+
+        foreach (var c in shuffle)
+            Console.WriteLine(c);
+        times++;
+        Console.WriteLine(times);
+    } while (!startingDeck.SequenceEquals(shuffle));
+    Console.WriteLine(times);
+}
+```
+
+Run again, and the inner shuffle is down to 30 queries. Run again with the outer shuffle
+and you'll see similar improvements. (It now executes 162 queries).
+
+Don't misinterpret this example by thinking that all queries should run eagerly. This is a particularly
+pathological case. In reality, some algorithms run much better using eager evaluation, and others run much
```

Comment:
It seems odd to have this example described as pathological since it seems like a fairly ordinary subject matter.  Maybe instead explain what makes this case unusual more precisely ?